### DISCUSSION TITLE
Inconsistent errors in the api/login and api/logout endpoints.

### DISCUSSION THREAD

**Discussion by aditi-mm** [gladly-team/next-firebase-auth#558] [Q&A]
I have an issue where the api/login and api/logout endpoints sometimes throw errors on my Vercel deployment. Even when I see an error on the console, the user is still able to login and access the home page of the application, but at times the home page does not fully load because I rely on tokens to pull from my api. This error happens about 1 in every 10 times I try and log in and out of my application. 

My api/login route and api/logout routes are exactly as specified in the documentation. I make sure to return a response in every case. Since the error does not happen every time, perhaps there is an issue with the API response not being awaited?

I saw a similar issue [here](https://github.com/gladly-team/next-firebase-auth/issues/557) but wasn't sure if I was seeing the same issue or there is something else I'm missing. 

Attached is a screenshot of the errors I see in the client's console. 
![image](https://user-images.githubusercontent.com/113858486/192028780-8e29b531-8129-4d52-b233-7c3a2b4db3c8.png)

**Comment by kmjennison** [comment]
Hi @aditi-mm, you'll need to provide more info for us to figure out what's going on here. That 500 from your login endpoint should have informative errors in server-side logs. Could you provide those? (Make sure you aren't try/catching without logging.)

In addition:

1. Have you set `onVerifyTokenError` and `onTokenRefreshError` in your config and checked for any helpful errors?
2. Have you set `debug: true` in your config and read through server-side and client-side debug logs for any helpful messages?
3. Have you tried the [example app](https://github.com/gladly-team/next-firebase-auth/tree/main/example) with your own Firebase credentials?

**Reply by aditi-mm** [reply]
Thank you for your prompt reply! I'm working on setting up more detailed logging right now, but in the meantime, here is what was logged on my server: 

Note: I can see some of my other api calls showing up in this error log, and perhaps the issue is that those other api calls are getting kicked off before the login api call can complete? Those api calls should only be kicked off when the user is logged in and accessing the homepage, however. I'm a little confused about how I can be accessing the user's id and tokens when the login api call hasn't completed. The same issue seems to be happening on logout, which means that the user's information would be temporarily available on the server even after they've logged out (just long enough to kick off the homepage api calls)?

![Screen Shot 2022-09-23 at 11 20 28 AM](https://user-images.githubusercontent.com/113858486/192032279-a8c2adb7-c24f-4ef8-abe5-42b40024add2.png)
![Screen Shot 2022-09-23 at 11 21 47 AM](https://user-images.githubusercontent.com/113858486/192032450-1207609a-5377-4749-8f26-69f3e7521bef.png)

Here is my login and logout api code:
```
initAuth()

const handler = async (
    req: NextApiRequest,
    res: NextApiResponse) => {
  try {
    await setAuthCookies(req, res)
  } catch (e) {
    return res.status(500).json({ error: 'Unexpected error.' })
  }
  return res.status(200).json({ success: true })
}

export default handler
```

```
initAuth()

const handler = async (
    req: NextApiRequest,
    res: NextApiResponse) => {
  try {
    await unsetAuthCookies(req, res)
  } catch (e) {
    return res.status(500).json({ error: 'Unexpected error.' })
  }
  return res.status(200).json({ success: true })
}

export default handler
```
